### ISSUE TITLE
Regular expression match not supported in memory backend?

### ISSUE THREAD

**Issue by matschaffer** [Netflix/atlas#157]
When running the sample data I can match things like `http://localhost:7101/api/v1/graph?q=name,DiscoveryStatus.*,:re,:sum` just fine, but when running the memory backend on my own data `:re` seems to only accept whole string matches.

For example `http://localhost:7101/api/v1/graph?s=e-8h&q=prefix,prd.core-testnet-001,:re` works but `http://localhost:7101/api/v1/graph?s=e-8h&q=prefix,prd.core-testnet,:re` (just trying to match the start of the string) doesn't. Also tried `.*` and `\d+` for the numbers but no success.

I haven't had a chance to work up a test case with fake data going into the memory backend yet so thought I'd open this up to ask if it's even expected to work.

**Comment by matschaffer** [comment]
Seems it's a little more subtle than that. I got a basic test case working locally and data of basically the same form on the same build of atlas works work on a local copy but returns "NO DATA" on a machine that's been running a few days.

Posting urls for sanity, 7101 is the local copy 7102 is a tunnel to the deployed version:

```
http://localhost:7101/api/v1/graph?q=prefix,prd.core-testnet-001,:re,stat,1_min_rate,:eq,name,ledger.ledger.close,:eq,:and,:and # works
http://localhost:7102/api/v1/graph?q=prefix,prd.core-testnet-001,:re,stat,1_min_rate,:eq,name,ledger.ledger.close,:eq,:and,:and # works

http://localhost:7101/api/v1/graph?q=prefix,prd.core-testnet,:re,stat,1_min_rate,:eq,name,ledger.ledger.close,:eq,:and,:and # works
http://localhost:7102/api/v1/graph?q=prefix,prd.core-testnet,:re,stat,1_min_rate,:eq,name,ledger.ledger.close,:eq,:and,:and # NO DATA
```

**Comment by matschaffer** [comment]
https://gist.github.com/matschaffer/3fbc53cbf2b7a5fd5f72 has the payload used for local testing

**Comment by brharrington** [comment]
Thanks for the detailed report, let me see if I can reproduce the issue.

**Comment by matschaffer** [comment]
I have logs if you need them, though the container's actually been running two weeks so it's ~100MB at this point.

**Comment by brharrington** [comment]
I think I see the problem. On your local copy do you have a single value for prefix and multiple on the deployed version?

If I'm right with the sample data you provided you should be able to reproduce locally using a query where the regex will match multiple values, e.g. I see history.publish.start and history.publish.delay so something like:

http://localhost:7101/api/v1/graph?q=name,history,:re,:sum

I should be able to get an updated release with a fix out later today.

As a side note, it is working internally and this would get noticed pretty quickly if broken. It looks like the changes we do to hook in some legacy backwards compatibility stuff internally avoids this issue. I'm going to try and minimize as many of the differences as I can for the 1.5 milestone, but it is taking a while.

**Comment by matschaffer** [comment]
I sure do. Nice find!

I'll press on with my atlas+heka-based alerting idea then. Looking forward to the patch!

**Comment by brharrington** [comment]
Fixed in 1.4.3.

**Comment by matschaffer** [comment]
Working great on 1.4.3. Thanks!
